src/go/k8s/pkg/admin/admin.go

```diff
@@ -53,11 +54,17 @@ func NewInternalAdminAPI(
 
 	adminInternalPort := adminInternal.Port
 
-	var urls []string
-	replicas := *redpandaCluster.Spec.Replicas
+	if len(ordinals) == 0 {
+		// Not a specific node, just go through all them
+		replicas := *redpandaCluster.Spec.Replicas
 
-	for i := int32(0); i < replicas; i++ {
-		urls = append(urls, fmt.Sprintf("%s-%d.%s:%d", redpandaCluster.Name, i, fqdn, adminInternalPort))
+		for i := int32(0); i < replicas; i++ {
+			ordinals = append(ordinals, i)
+		}
+	}
+	urls := make([]string, 0, len(ordinals))
+	for _, on := range ordinals {
+		urls = append(urls, fmt.Sprintf("%s-%d.%s:%d", redpandaCluster.Name, on, fqdn, adminInternalPort))
```

Comment:
The `on` in this string: it's a k8s-specific pod index that's guaranteed to be 0..n, not a node ID, right?  (if it was a node ID then it would not be safe to assume they were consecutive from zero)

Reply:
Yes, pod ordinal